### PR TITLE
ZCS-11154: GetAppointmentIdsSinceResponse not populating deleted item ids

### PR DESCRIPTION
**Pull request by shrutig0510**
Fixed issue with deleted item ids in GetAppointmentIdsSinceResponse.
It was only returning all modified item ids

### CODE REVIEW

File: store/src/java/com/zimbra/cs/db/DbMailItem.java [hunk 1]
```diff
@@ -4291,12 +4294,14 @@ private static PreparedStatement calendarItemStatement(DbConnection conn, String
...
         if (folderSpecified) {
             stmt.setInt(pos++, folderId);
+            stmt.setString(pos++, "%:"+folderId);
         }
         if (excludeFolderIds != null) {
             for (int id : excludeFolderIds) {
                 stmt.setInt(pos++, id);
             }
         }
+
```
Comment (abhishekdeshmukh2209) [inline store/src/java/com/zimbra/cs/db/DbMailItem.java:4304]: extra line

Comment (shrutig0510) [reply store/src/java/com/zimbra/cs/db/DbMailItem.java]: Done

**Review by abhishekdeshmukh2209 — APPROVED** [review]
LGTM

**Review by zimsuchitgupta — APPROVED** [review]
LGTM

**Review by desouzas — APPROVED** [review]

File: store/src/java/com/zimbra/cs/db/DbMailItem.java [hunk 2]
```diff
@@ -4281,8 +4283,9 @@ private static PreparedStatement calendarItemStatement(DbConnection conn, String
                 " FROM " + getCalendarItemTableName(mbox, "ci") + ", " + getMailItemTableName(mbox, "mi") +
                 " WHERE mi.id = ci.item_id" + lastSyncConstraint + " AND mi." + typeConstraint +
                 (DebugConfig.disableMailboxGroups? "" : " AND ci.mailbox_id = ? AND mi.mailbox_id = ci.mailbox_id") +
-                (folderSpecified ? " AND folder_id = ?" : "") + excludeFolderPart + orderByConstraint);
-
+                (folderSpecified ? " AND ( folder_id = ? OR (folder_id = '"+ FolderConstants.ID_FOLDER_TRASH + "' AND prev_folders LIKE ? ))" : "") +
```
[the remark this answers is not in the document]
Comment (shrutig0510) [reply store/src/java/com/zimbra/cs/db/DbMailItem.java]: Done

**Review by rcyarrapothu — APPROVED** [review]

**Pull request merged**